File: 2025/4주차/BOJ4948_베르트랑_공준.py
import sys
input = sys.stdin.readline

MAX = 246912
# 미리 소수 여부를 구해둠
is_prime = [True] * (MAX + 1)
is_prime[0] = is_prime[1] = False

for i in range(2, int(MAX ** 0.5) + 1):
    if is_prime[i]:
        for j in range(i * i, MAX + 1, i):
            is_prime[j] = False

while True:
    num = int(input())
    if num == 0:
        break
    # num+1부터 2*num까지 소수 개수 세기
    count = sum(1 for i in range(num + 1, 2 * num + 1) if is_prime[i])
    print(count)

File: 2025/4주차/BOJ4948_베르트랑 공준.cpp
#include <iostream>
#include <cmath>
using namespace std;
const int MAX = 246912;  //123456의 2배

bool isPrime[MAX + 1];

int main(){
    ios_base::sync_with_stdio(0);
    cin.tie(0); cout.tie(0);

    fill_n(isPrime, MAX + 1, true);
    isPrime[0] = isPrime[1] = false;

    for (int i = 2; i * i <= MAX; i++) {
        if (isPrime[i]) {
            for (int j = i * i; j <= MAX; j += i) {
                isPrime[j] = false;
            }
        }
    }
    
    while (true) {
        int n;
        cin >> n;
        if (n == 0) break;

        int count = 0;
        for (int i = n + 1; i <= 2 * n; ++i) {
            if (isPrime[i]) ++count;
        }

        cout << count << "\n";
    }

    return 0;

}
